crates/ethereum/evm/src/execute.rs

```diff
@@ -227,7 +227,12 @@ where
             let deposit_requests =
                 crate::eip6110::parse_deposits_from_receipts(&self.chain_spec, receipts)?;
 
-            let mut requests = Requests::new(vec![deposit_requests]);
+            let mut requests = Requests::default();
+
+            if !deposit_requests.is_empty() {
+                requests.push_request(core::iter::once(0).chain(deposit_requests).collect());
```

Comment:
yep, before we were doing this in `Requests` because they were expected to always contain 3 elements, but empty values are now should be removed so this needs to be done by caller